src/pip/_internal/exceptions.py

```diff
@@ -149,17 +148,17 @@ def __init__(self, *, package: str, reason: str) -> None:
 
 
 class NoneMetadataError(PipError):
```

Comment:
For a follow up, we should probably give this a more meaningful name -- `DeclaredMetadataFileIsMissingError` or something similar.

Reply:
FWIW this exception never actually makes sense; it’s caught in exactly two places, one just default to an empty string, and the other is the outmost catch-all block that prints the error message. It will probably go away entirely when the importlib.metadata backend gets implemented.

Reply:
Well, if I remember correctly, the situation which this was originally raised in was something weird, like a metadata file that has a truth has_metadata but get_metadata returned None. I'm not quite sure how this could happen, TBH. I do remember seeing this exception get raised in another situation in one of these porting PRs, which might have changed the meaning of it?

If we're sure that this will never happen with importlib metadata, then yea, this can probably go away.

Reply:
> the situation which this was originally raised in was something weird, like a metadata file that has a truth has_metadata but get_metadata returned None. I'm not quite sure how this could happen, TBH.

That’s correct, and the exception _can_ happen for permission reasons. But the thing is that this discrepency isn’t really necessary due to how the exception is handled. And this also does not happen in importlib.metadata because has_metadata does not even exist in the API—the only way to know if a metadata file exists in importlib.metadata is try to read it.